### PR TITLE
Add leaderboard reset button

### PR DESCRIPTION
**Pull request by rparrett**
It was a pretty amazing journey getting set up to develop and test this tiny changeset.

It's pretty basic. UI doesn't update when resetting.

Would be great to have this functionality when the bevy jam voting period begins. It's a bit of an ordeal to rebuild and re-upload to swap to a fresh leaderboard.

### CODE REVIEW

File: jornet-server/src/domains/score.rs [hunk 1]
```diff
@@ -105,6 +112,7 @@ pub(crate) fn score() -> impl HttpServiceFactory {
         .wrap(cors)
         .route("{leaderboard_id}", web::post().to(save_score))
         .route("{leaderboard_id}", web::get().to(get_scores))
+        .route("{leaderboard_id}", web::delete().to(delete_scores))
```
Comment (mockersf) [inline jornet-server/src/domains/score.rs:115]: putting the new endpoint here makes it reachable without authentication

You would need to put it in a scope that checks the bearer (for example https://github.com/vleue/jornet/blob/main/jornet-server/src/domains/leaderboard.rs#L58), and check that the leaderboard owner is the same as the token user

**Comment by rparrett** [comment]
I moved it to `DELETE /leaderboards/{leaderboard_id}/scores` under the existing scope that validates tokens in `leaderboards.rs`.

I also added leaderboard ownership verification and tested both the Ok and Unauthorized responses.

**Review by mockersf — APPROVED** [review]

**Comment by mockersf** [comment]
looks good! do you want to move it out of draft?

**Comment by mockersf** [comment]
deployed 👍

**Pull request merged**